test.sh

```diff
@@ -24,12 +30,92 @@ function prepareOutputDir() {
   mkdir -p output
 }
 
-function createTestEnvironment() {
+# Build DAP test container & start the cluster
+function createDAPTestEnvironment() {
+  # docker-compose build --pull client-dap cuke-master postgres test-dap
+  docker-compose up -d client-dap cuke-master postgres test-dap
+
+  # Delay to allow time for conjur to come up
+  echo 'Waiting for conjur server to be healthy'
+  docker-compose run --rm test ./wait_for_server.sh
+}
+
+function loadDAPTestPolicy() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Loading DAP test policy"
+  echo '------------------------------------------------------------'
+
+  # get DAP container id
+  dap_client_cid=$(docker-compose ps -q client-dap)
+
+  ssl_cert=$(docker-compose exec cuke-master cat /opt/conjur/etc/ssl/conjur.pem)
+  export CONJUR_SSL_CERTIFICATE=$(docker-compose exec cuke-master cat /opt/conjur/etc/ssl/conjur.pem)
+
+  docker-compose up -d client-dap
+  docker-compose exec client-dap conjur authn login -u admin -p secret
+
+  # copy test-policy into a /tmp/test-policy within the client container
+  docker cp test-policy ${dap_client_cid}:/tmp
+
+  docker-compose exec client-dap bash -c "conjur policy load root /tmp/test-policy/root.yml"
+}
+
+function initializeDAPCert () {
+  dap_client_cid=$(docker-compose ps -q client-dap)
+    # get the pem file from conjur server
+  CONJUR_ACCOUNT="cucumber"
+  CONJUR_PROXY="https://cuke-master"
+
+#  NOT WORKING
+  echo "remove old pem file"
+  rm -rf /test-cert/*
+
+  echo "fetch pem file from proxy https server"
+  exec_command='echo yes | conjur init -u '${CONJUR_PROXY}' -a '${CONJUR_ACCOUNT}' > tmp.out 2>&1'
```

Comment:
This string will not properly evaluate since you use variables inside it but it is surrounded by single quotes (`'`). You should convert this string to use double-quotes (`"`).

Reply:
What is the reasoning for putting output into tmp.out and printing it out instead of letting docker print it out when running the command?

Reply:
As a sidenote (also mentioned elsewhere), when you run just `bash -c` you are (like the first line in this file) ignoring any errors that occur in the running of the command. Think about if that's desired here or not.